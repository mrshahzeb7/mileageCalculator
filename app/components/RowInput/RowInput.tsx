import React, {ReactElement} from 'react';
import {
  ImageSourcePropType,
  StyleProp,
  StyleSheet,
  View,
  ViewStyle,
} from 'react-native';
import {colors, iconSizes, spacing} from '../../theme';
import {Icon} from '../Icon/Icon';

export interface RowInputProps {
  icon: ImageSourcePropType;
  ChildOne: ReactElement;
  ChildTwo?: ReactElement;
  style?: StyleProp<ViewStyle>;
}

export const RowInput: React.FC<RowInputProps> = ({
  icon,
  ChildOne,
  ChildTwo,
  style,
}) => {
  return (
    <View style={[styles.container, style]}>
      <Icon
        style={styles.icon}
        size={iconSizes.small}
        color={colors.primaryButton}
        icon={icon}
      />

      {!!ChildOne && ChildOne}
      {!!ChildTwo && ChildTwo}
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: spacing.horizontal.medium,
  },
  icon: {
    marginHorizontal: spacing.small,
  },
});
