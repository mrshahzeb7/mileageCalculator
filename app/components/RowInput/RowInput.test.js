import 'react-native';
import React from 'react';
import {RowInput} from './RowInput';
import renderer from 'react-test-renderer';

describe('RowInput', () => {
  it(' RowInput renders correctly', () => {
    const tree = renderer.create(<RowInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
