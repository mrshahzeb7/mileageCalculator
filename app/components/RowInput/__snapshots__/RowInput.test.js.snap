// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowInput  RowInput renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingRight": 37.5,
      },
      undefined,
    ]
  }
>
  <View>
    <Image
      style={
        [
          {
            "resizeMode": "contain",
          },
          {
            "tintColor": "#93b2fd",
          },
          {
            "height": 50.02435032061885,
            "width": 50.02435032061885,
          },
          {
            "marginHorizontal": 30.595932917809698,
          },
        ]
      }
    />
  </View>
</View>
`;
